diff --git a/Dockerfile b/Dockerfile
index c1dbf6a..967e785 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,6 @@
 FROM nginx:1.19
-COPY ./build /var/www
+COPY ./build-sa /var/www/superadmin
+COPY ./build-bm /var/www/management
 COPY nginx.conf /etc/nginx/nginx.conf
 EXPOSE 80
 ENTRYPOINT ["nginx", "-g", "daemon off;"]
diff --git a/Makefile b/Makefile
index 109762b..108366f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,12 @@
 TAG=latest
-REPO=registry.gitlab.com/clink1/building-management-apps-web/web:${TAG}
+REPO=registry.gitlab.com/clink1/building-management-apps-web/adminweb:${TAG}
 
 npmbuild:
-	rm -rf build
-	npm run build
+	rm -rf build build-sa build-bm
+	REACT_APP_DEFAULT_ROLE=sa npm run build
+	mv build build-sa
+	REACT_APP_DEFAULT_ROLE=bm npm run build
+	mv build build-bm
 
 build: npmbuild
 	docker build -t ${REPO} .
@@ -13,7 +16,11 @@ push:
 
 start: 
 	echo "Running dockerized instance of web apps, access at localhost:8080/"
-	docker run -it -p 8080:80 ${REPO}
+	docker run -it -p 80:80 ${REPO}
 
+test:
+	docker build -t appsweb .
+	docker stop appsweb; docker rm appsweb;
+	docker run -dt -p 80:80 --name appsweb appsweb:latest
 
 .PHONY: build
diff --git a/nginx.conf b/nginx.conf
index d56813d..14671f6 100644
--- a/nginx.conf
+++ b/nginx.conf
@@ -11,51 +11,80 @@ events {
 
 
 http {
-    # what times to include
-    include       /etc/nginx/mime.types;
-    # what is the default one
-    default_type  application/octet-stream;
-
     # Sets the path, format, and configuration for a buffered log write
     log_format compression '$remote_addr - $remote_user [$time_local] '
         '"$request" $status $upstream_addr '
         '"$http_referer" "$http_user_agent"';
 
+    index index.html;
+
+    server {
+        server_name admin.yipy.id;
+
+        # save logs here
+        access_log /var/log/nginx/admin.yipy.id.log compression;
+        error_log /var/log/nginx/admin.yipy.id.error.log debug;
+
+        # where the root here
+        root /var/www/superadmin;
+
+        # if served from here go to sa
+        location = / {
+            return 301 $scheme://$host/sa;
+        }
+
+        location ~* /bm/(.*)$ {
+            return 301 $scheme://$host/sa/$1;
+        }
+
+        location ~ \.css {
+            add_header  Content-Type    text/css;
+        }
+
+        location ~ \.js {
+            add_header  Content-Type    application/x-javascript;
+        }
+
+        location ~* /(.*)$ {
+            try_files $uri $uri/ /index.html /404.html =404;
+        }
+    }
+
     server {
-        # listen on port 80
-        listen 80;
+        server_name management.yipy.id;
+
         # save logs here
-        access_log /var/log/nginx/access.log compression;
+        access_log /var/log/nginx/management.yipy.id.log compression;
+        error_log /var/log/nginx/management.yipy.id.error.log debug;
 
         # where the root here
-        root /var/www;
-        # what file to server as index
-        index index.html index.htm;
+        root /var/www/management;
 
-        location / {
-            # First attempt to serve request as file, then
-            # as directory, then fall back to redirecting to index.html
-            try_files $uri $uri/ /index.html;
+        # if served from here go to bm
+        location = / {
+            return 301 $scheme://$host/bm;
         }
 
-        # Media: images, icons, video, audio, HTC
-        location ~* \.(?:jpg|jpeg|gif|png|ico|cur|gz|svg|svgz|mp4|ogg|ogv|webm|htc)$ {
-          expires 1M;
-          access_log off;
-          add_header Cache-Control "public";
+        location ~* /sa/(.*)$ {
+            return 301 $scheme://$host/bm/$1;
         }
 
-        # Javascript and CSS files
-        location ~* \.(?:css|js)$ {
-            try_files $uri =404;
-            expires 1y;
-            access_log off;
-            add_header Cache-Control "public";
+        location ~ \.css {
+            add_header  Content-Type    text/css;
         }
 
-        # Any route containing a file extension (e.g. /devicesfile.js)
-        location ~ ^.+\..+$ {
-            try_files $uri =404;
+        location ~ \.js {
+            add_header  Content-Type    application/x-javascript;
+        }
+
+        location ~* /(.*)$ {
+            try_files $uri $uri/ /index.html /404.html =404;
+        }
+    }
+
+    server {
+        location /health {
+            return 200 'ok';
         }
     }
 }
diff --git a/src/App.js b/src/App.js
index a0fc7f9..556c04f 100644
--- a/src/App.js
+++ b/src/App.js
@@ -7,6 +7,7 @@ import {
 } from "react-router-dom";
 import { PersistGate } from "redux-persist/integration/react";
 import { Provider, useSelector } from "react-redux";
+import { defaultRole } from "./settings";
 
 import "./App.css";
 import "./features/auth/styles.css";
@@ -86,35 +87,31 @@ function OTPRoute({ children, ...other }) {
 }
 
 function AppRoute() {
-  const { role } = useSelector((state) => state.auth);
-
   return (
     <Router>
       <div className="App scroller" id="App">
         <Switch>
-          <Redirect exact from="/" to={"/" + role} />
-          {role === 'bm' && <Redirect from="/sa" to={"/bm"} />}
-          {role === 'sa' && <Redirect from="/bm" to={"/sa"} />}
+          <Redirect exact from="/" to={"/" + defaultRole } />
           {/* superadmin routes */}
-          <Route path="/sa/login">
+          { defaultRole === 'sa' && <Route path="/sa/login">
             <Login role="sa" />
-          </Route>
-          <OTPRoute path="/sa/otp">
+          </Route>}
+          { defaultRole === 'sa' && <OTPRoute path="/sa/otp">
             <OTP role="sa" />
-          </OTPRoute>
-          <MainRoute path="/sa">
-            <SA />
-          </MainRoute>
+          </OTPRoute>}
+          { defaultRole === 'sa' && <MainRoute path="/sa">
+             <SA />
+          </MainRoute> }
           {/* bm routes */}
-          <Route path="/bm/login">
+          { defaultRole === 'bm' && <Route path="/bm/login">
             <Login role="bm" />
-          </Route>
-          <OTPRoute path="/bm/otp">
+          </Route>}
+          { defaultRole === 'bm' && <OTPRoute path="/bm/otp">
             <OTP role="bm" />
-          </OTPRoute>
-          <MainRoute path="/bm">
+          </OTPRoute>}
+          { defaultRole === 'bm' && <MainRoute path="/bm">
             <BM />
-          </MainRoute>
+          </MainRoute>}
           <Route path="*">
             <NotFound />
           </Route>
diff --git a/src/settings.js b/src/settings.js
index cf1c4d1..e29ca74 100644
--- a/src/settings.js
+++ b/src/settings.js
@@ -1,3 +1,5 @@
+
+
 export const endpointBase = 'https://centratama.okbabe.technology';
 
 export const endpointAsset = endpointBase + '/clink-assets';
@@ -12,6 +14,8 @@ export const endpointAds = endpointBase + '/clink-advertisement';
 
 export const mainColor = '#2f78e9';
 
+export const defaultRole = process.env.REACT_APP_DEFAULT_ROLE || 'sa';
+
 export const taskStatusColor = {
     'created': 'light',
     'assigned': 'success',
